WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductFromImage/AddProductFromImageEligibilityChecker.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+import Yosemite
+import protocol Experiments.FeatureFlagService
+
+/// Protocol for checking "add product from image" eligibility for easier unit testing.
+protocol AddProductFromImageEligibilityCheckerProtocol {
+    /// Checks if the user is eligible to participate in the A/B experiment.
+    func isEligibleToParticipateInABTest() -> Bool
+
+    /// Checks if the user is eligible for the "add product from image" feature.
+    func isEligible() -> Bool
+}
+
+/// Checks the eligibility for the "add product from image" feature.
+final class AddProductFromImageEligibilityChecker: AddProductFromImageEligibilityCheckerProtocol {
+    private let stores: StoresManager
+    private let featureFlagService: FeatureFlagService
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.stores = stores
+        self.featureFlagService = featureFlagService
+    }
+
+    func isEligibleToParticipateInABTest() -> Bool {
+        stores.sessionManager.defaultSite?.isWordPressComStore == true
+    }
+
+    func isEligible() -> Bool {
+        guard isEligibleToParticipateInABTest() else {
+            return false
+        }
```

Comment:
Same as in https://github.com/woocommerce/woocommerce-ios/pull/10184#discussion_r1255851088

Reply:
In the above comment you mentioned this:

> Then, only the treatment variant will enter the second isEligible path for the new flow. 

That means the check for `isEligibleToParticipateInABTest` is redundant here right?

Reply:
If you meant if `isEligibleToParticipateInABTest` is necessary:

For the treatment variant, we still need to track the exposure event to start the A/B experiment.

There are users (using non-WPCOM stores) who aren't eligible for the A/B experiments who can enter this function, so it's necessary to only track the exposure event only for `isEligibleToParticipateInABTest`.

---

If you meant why checking `isEligibleToParticipateInABTest` inside `isEligible` since it was checked in the outer condition:

I can leave this out since we have to track the exposure event for both variants before this, I just didn't want to make it look like this isn't part of the eligibility for the feature. The check is trivial (not async or something computationally intensive) so I thought I can keep it here. Otherwise, we have to have some way to ensure that the logic in `isEligibleToParticipateInABTest` has to be transferred to `isEligible` when removing the A/B experiment.

Lemme know if I'm missing something or you have any other suggestions.

Reply:
Thanks for the explanation! I was wondering because the checks overlap, but your explanation helps me understand the idea behind this. If possible, it would be great to add the explanation to the code in a comment to avoid misunderstanding.

Reply:
Added a comment about this in https://github.com/woocommerce/woocommerce-ios/pull/10208/commits/81ebd4cd54dbe34148f149c393122f7829ced68e